### DISCUSSION TITLE
Error 500

### DISCUSSION THREAD

**Discussion by CodeMouse92** [horilla-opensource/horilla#130] [Q&A]
I'm running Ubuntu 22.04, Python 3.10, Django 4.2.11, and the latest `master` of Horilla. I have configured the server as best as I can tell following the instructions, and then try to serve it over Apache2 as shown below.

By the way, this worked once before, but after recloning Horilla to try and get some bugfixes (the first attempt to install this was a nightmare too), this no longer works.

Unfortunately, I only see "Server Error (500)", and the Apache error logs show me only this:

```
[Fri Mar 22 15:00:16.172940 2024] [wsgi:error] [pid 1855169] /opt/horilla/venv/lib/python3.10/site-packages/pandas/__init__.py:14: UserWarning: NumPy was imported from a Python sub-interpreter but NumPy does not properly support sub-interpreters. This will likely work for most users but might cause hard to track down issues or subtle bugs. A common user of the rare sub-interpreter feature is wsgi which also allows single-interpreter mode.
[Fri Mar 22 15:00:16.172978 2024] [wsgi:error] [pid 1855169] Improvements in the case of bugs are welcome, but is not on the NumPy roadmap, and full support may require significant effort to achieve.
[Fri Mar 22 15:00:16.172981 2024] [wsgi:error] [pid 1855169]   __import__(_dependency)
```

I subsequently fixed that following some StackOverflow insight (https://stackoverflow.com/a/70903486), so the warning disappears, but Error 500 persists.

Is there logging somewhere in Horilla? What am I overlooking?

---

Here's the (scrubbed) environment file:

```env
DEBUG=False

SECRET_KEY=REPLACE_ME

ALLOWED_HOSTS=eco.mousepawmedia.com,localhost:8000

DB_ENGINE=django.db.backends.postgresql
DB_NAME=horilla_production
DB_USER=horilla
DB_PASSWORD=REPLACE_ME
DB_HOST=localhost
DB_PORT=5432
```

I set up Apache like this:

```bash
sudo apt install -y libapache2-mod-wsgi-py3
sudo a2enmod wsgi
sudo vim /etc/apache2/sites-available/eco.conf
```

```
<VirtualHost *:80>
    ServerName eco.mousepawmedia.com
    ServerSignature Off

    RewriteEngine on
    RewriteCond %{HTTPS} !=on
    RewriteRule .* https://%{SERVER_NAME}%{REQUEST_URI} [NE,R,L]
</VirtualHost>

<VirtualHost *:443>
    ServerName eco.mousepawmedia.com
    ServerSignature Off

    SSLEngine On
    SSLProtocol TLSv1 TLSv1.1 TLSv1.2
    SSLProxyEngine on
    SSLHonorCipherOrder On
    SSLCipherSuite "EECDH+ECDSA+AESGCM:EECDH+aRSA+AESGCM:EECDH+ECDSA+SHA256:EECDH+aRSA+SHA256:EECDH+ECDSA+SHA384:EECDH+ECDSA+SHA256:EECDH+aRSA+SHA384:EDH+aRSA+AESGCM:EDH+aRSA+SHA256:EDH+aRSA:EECDH:!aNULL:!eNULL:!MEDIUM:!LOW:!3DES:!MD5:!EXP:!PSK:!SRP:!DSS:!RC4:!SEED"
    Header set Strict-Transport-Security "max-age=31536000"

    SSLCertificateFile /etc/letsencrypt/live/eco.mousepawmedia.com/fullchain.pem
    SSLCertificateKeyFile /etc/letsencrypt/live/eco.mousepawmedia.com/privkey.pem

    Alias /static /opt/horilla/staticfiles
    <Directory /opt/horilla/staticfiles/>
        Require all granted
    </Directory>

    <Directory /opt/horilla/horilla>
        <Files wsgi.py>
            Require all granted
        </Files>
    </Directory>

    WSGIDaemonProcess horilla python-home=/opt/horilla/venv python-path=/opt/horilla
    WSGIProcessGroup horilla
    WSGIScriptAlias / /opt/horilla/horilla/wsgi.py

    ErrorLog ${APACHE_LOG_DIR}/horilla_error.log
</VirtualHost>
```

```bash
sudo a2ensite eco
sudo systemctl restart apache2
```

I've documented the entire setup process, if anyone wants to see the rest.

**Comment by horilla-opensource** [comment]
> `[Fri Mar 22 15:00:16.172940 2024] [wsgi:error] [pid 1855169] /opt/horilla/venv/lib/python3.10/site-packages/pandas/__init__.py:14: UserWarning: NumPy was imported from a Python sub-interpreter but NumPy does not properly support sub-interpreters. This will likely work for most users but might cause hard to track down issues or subtle bugs. A common user of the rare sub-interpreter feature is wsgi which also allows single-interpreter mode.`

Hi @CodeMouse92 ,
From the log that you have shared, I don't think its issue with the Horilla code base, rather an issue of the Numpy package. Also to mention, I don't think its issue with the Apache's configurations too.
May be you can check the status of the service you have created for horilla.
And for the log, if you have added the horilla in ubuntu as service file, you can configure the log file inside the system file.
Below is an example to get the log file running for getting a more understanding of the error.
`[Unit]
Description=Horilla Main

[Service]
User=root
Group=www-data
WorkingDirectory=/home/ubuntu/Horilla_Main/
ExecStart=/usr/bin/python3 manage.py runserver 0.0.0.0:8000
StandardOutput=file:<path _to_your_logfile>/horilla_main.log
[Install]
WantedBy=multi-user.target
`
Above is a basic service file configuration for running Horilla, and there for the log you can add the StandardOutput to your log file location.

**Comment by CodeMouse92** [comment]
I don't know that I can set up the service like that, as Apache2 is running Horilla via WSGI, rather than as a reverse proxy. (WSGI seems to be rubbish at logging.) I couldn't find any good instructions on properly configuring a reverse proxy for Apache2, although I'd be game if there are.

**Comment by horilla-opensource** [comment]
Hi @CodeMouse92 ,
Currently we don't have any docs related to the configuration using apache2.
We value your concern and are working on providing you with a solution for the same.

With Regards,
**Team Horilla**
